lib/csv_helper/loader.rb

```diff
@@ -10,13 +10,21 @@ module Loader
     }.freeze
 
     def load(file, options = {})
-      delete_all
-      load_csv_file(file, options)
+      klass.transaction do
+        delete_all
+        load_csv_file(file, options)
+      end
+    rescue StandardError => e
+      "Error #{e.message}"
```

Comment:
Are you sure `load` should return a string when there's an exception?  Looking in the uploads_controller, I think a string would cause an error when `data.ids` is called in `call_load`.  I think it would probably be okay to let the error just bubble up since the uploads controller has some rescues.  Might be good to add a test for this issue in the uploader_controller spec.  This would probably also mean changing the test for this to expect the block to `raise_error` and then instead of using the change matcher just wrap it yourself with `initial_count = described_class.count` and after the error using `expect(described_class.count).to eq initial_count`